pool/src/types.rs

```diff
@@ -129,7 +129,7 @@ pub struct PoolEntry {
 }
 
 /// The possible states a pool entry can be in.
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Clone, Copy, Debug, PartialEq)]
```

Comment:
I'm never clear on what _exactly_ this does... this allows us to get rid of some calls to `clone()` above but we're not actually gaining anything performance wise?

Reply:
Correct, it changes assignment semantics from move to copy, so right hand value could be used after assignment (like in normal languages). Usually we achieve that with explicit `.clone()`. `Copy` is implicit and can't be overloaded, it's always bit-wise copy. Rust documentation makes a bit controversial claim `Generally speaking, if your type can implement Copy, it should.` https://doc.rust-lang.org/std/marker/trait.Copy.html which mean that you should have as least Rust as possible (move  assignment semantics is the most Rust-specific behavior, cornerstone of borrow/ownership thing)